tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1462,3 +1462,257 @@ def test_metric_computation_handles_absent_labels():
+
+    mlflow.sklearn.autolog()
+
+    model = sklearn.linear_model.LogisticRegression(solver="saga", max_iter=100, random_state=0)
+    X, y = get_iris()
+
+    with mlflow.start_run() as run:
+        model.fit(X, y)
+
+        eval1_X, eval1_y = X[0::3], y[0::3]
+        eval2_X, eval2_y = X[1::3], y[1::3]
+
+        pred1_y = model.predict(X=eval1_X)
+        pred2_y = model.predict(eval2_X)
+
+        r2_score_data1 = sklmetrics.r2_score(eval1_y, pred1_y)
+        mean_squared_error_data1 = sklmetrics.mean_squared_error(eval1_y, pred1_y)
+        r2_score_data2 = sklmetrics.r2_score(eval2_y, pred2_y)
+        lor_score_data1 = model.score(eval1_X, eval1_y)
+        mean_squared_error_data2 = sklmetrics.mean_squared_error(eval2_y, pred2_y, squared=False)
+
+        scorer1 = sklmetrics.make_scorer(sklmetrics.mean_squared_error, squared=False)
+        mean_squared_error3_data2 = scorer1(model, eval2_X, eval2_y)
+
+        eval1_X, eval1_y = eval1_X.copy(), eval1_y.copy()
+        # In metric key, it will include dataset name as "eval1_X-2"
+        lor_score_data1_2 = model.score(eval1_X, eval1_y)
+
+        # In metric key, it will include dataset name as "unknown_dataset"
+        lor_score_data1_3 = model.score(eval1_X.copy(), eval1_y.copy())
+
+        metric_info = load_json_artifact("metric_info.json")
+
+    run_id = run.info.run_id
+    params, metrics, tags, artifacts = get_run_data(run_id)
+    post_training_metrics = {k: v for k, v in metrics.items() if not k.startswith('training_')}
+
+    assert post_training_metrics == {
+        'r2_score_eval1_X': r2_score_data1,
+        'mean_squared_error_eval1_X': mean_squared_error_data1,
+        'r2_score-2_eval2_X': r2_score_data2,
+        'LogisticRegression_score_eval1_X': lor_score_data1,
+        'mean_squared_error-2_eval2_X': mean_squared_error_data2,
+        'mean_squared_error-3_eval2_X': mean_squared_error3_data2,
+        'LogisticRegression_score-2_eval1_X-2': lor_score_data1_2,
+        'LogisticRegression_score-3_unknown_dataset': lor_score_data1_3,
+    }
+
+    assert metric_info == {
+        'LogisticRegression_score-2_eval1_X-2': 'LogisticRegression.score(X=eval1_X, y=eval1_y)',
+        'LogisticRegression_score-3_unknown_dataset': 'LogisticRegression.score(X=<ndarray>, y=<ndarray>)',
+        'LogisticRegression_score_eval1_X': 'LogisticRegression.score(X=eval1_X, y=eval1_y)',
+        'mean_squared_error-2_eval2_X': 'mean_squared_error(y_true=eval2_y, y_pred=pred2_y, squared=False)',
+        'mean_squared_error-3_eval2_X': 'mean_squared_error(y_true=eval2_y, y_pred=y_pred, squared=False)',
+        'mean_squared_error_eval1_X': 'mean_squared_error(y_true=eval1_y, y_pred=pred1_y)',
+        'r2_score-2_eval2_X': 'r2_score(y_true=eval2_y, y_pred=pred2_y)',
+        'r2_score_eval1_X': 'r2_score(y_true=eval1_y, y_pred=pred1_y)'
+    }
+
+    mlflow.sklearn.autolog(disable=True)
+
+    # Test patched methods generate the same results with unpatched methods.
```

Comment:
Can we clarify that `sklmetrics.mean_squared_error` is unpatched because it's imported before the call to `mlflow.sklearn.autolog()`?

Reply:
I think we need to import a function, not a module to obtain the original function.

```python
from sklearn.metrics import accuracy_score
from sklearn import metrics
import sklearn.metrics

# Patch
sklearn.metrics.accuracy_score = lambda *args, **kwargs: None

print(id(accuracy_score))
print(id(metrics.accuracy_score))
print(id(sklearn.metrics.accuracy_score))

print(accuracy_score([0, 1], [1, 1]))
print(metrics.accuracy_score([0, 1], [1, 1]))
print(sklearn.metrics.accuracy_score([0, 1], [1, 1]))
```

This gives:

```
140228484179216
140231442587296
140231442587296
0.5
None
None
```

Reply:
```
from sklearn import metrics

mlflow.sklearn.autolog()
metrics.accuracy_score(...) # now this method has been patched.

mlflow.sklearn.autolog(disable=True)
metrics.accuracy_score(...) # now this method `metrics.accuracy_score` has been restored to be the original method.

```

Reply:
Or we can simply do:

```python
mlflow.sklearn.autolog

assert metrics.accuracy_score([0, 0], [0, 1]) == 0.5
```

Reply:
Besides metric apis, we need also check the `model.score` , `model.predict` return value not affected by autologging patch. current test code is simple for these cases.